<a href="https://colab.research.google.com/github/hstevans/Cirrhosis-Stroke-Predictions--Machine-Learning--Classification-Models/blob/main/Cirrhosis_Stroke_Predictions_Machine_Learning_Classification_Models_Part_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##***Cirrhosis-Stroke-Predictions--Machine-Learning--Classification-Models*** 

By: Hollyann Stevans


Assignment:

Choose 2 datasets and answers questions about each.

###**Importing Libraries**

In [2]:
import pandas as pd

###**Dataset 1**

###Uploading Data:

In [8]:
data1 = pd.read_csv("/content/drive/MyDrive/Colab Notebooks/cirrhosis.csv")
data1.head()

ID  N_Days Status             Drug    Age Sex Ascites Hepatomegaly Spiders  \
0   1     400      D  D-penicillamine  21464   F       Y            Y       Y   
1   2    4500      C  D-penicillamine  20617   F       N            Y       Y   
2   3    1012      D  D-penicillamine  25594   M       N            N       N   
3   4    1925      D  D-penicillamine  19994   F       N            Y       Y   
4   5    1504     CL          Placebo  13918   F       N            Y       Y   

  Edema  Bilirubin  Cholesterol  Albumin  Copper  Alk_Phos    SGOT  \
0     Y       14.5        261.0     2.60   156.0    1718.0  137.95   
1     N        1.1        302.0     4.14    54.0    7394.8  113.52   
2     S        1.4        176.0     3.48   210.0     516.0   96.10   
3     S        1.8        244.0     2.54    64.0    6121.8   60.63   
4     N        3.4        279.0     3.53   143.0     671.0  113.15   

   Tryglicerides  Platelets  Prothrombin  Stage  
0          172.0      190.0         12.2    4.0  
1           88.0      221.0         10.6    3.0  
2           55.0      151.0         12.0    4.0  
3           92.0      183.0         10.3    4.0  
4           72.0      136.0         10.9    3.0

In [9]:
data1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 20 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   ID             418 non-null    int64  
 1   N_Days         418 non-null    int64  
 2   Status         418 non-null    object 
 3   Drug           312 non-null    object 
 4   Age            418 non-null    int64  
 5   Sex            418 non-null    object 
 6   Ascites        312 non-null    object 
 7   Hepatomegaly   312 non-null    object 
 8   Spiders        312 non-null    object 
 9   Edema          418 non-null    object 
 10  Bilirubin      418 non-null    float64
 11  Cholesterol    284 non-null    float64
 12  Albumin        418 non-null    float64
 13  Copper         310 non-null    float64
 14  Alk_Phos       312 non-null    float64
 15  SGOT           312 non-null    float64
 16  Tryglicerides  282 non-null    float64
 17  Platelets      407 non-null    float64
 18  Prothrombi

**1. Source of data:** https://www.kaggle.com/datasets/fedesoriano/cirrhosis-prediction-dataset

**2. Brief description of data:**

  This dataset is used to predict what stage of liver cirrhosis a patient has (target)
  based on information about the patient (features) such as age, gender, cholesterol, edma, etc

**3. What is the target?** 

'stage'

**4. What does one row represent? (A person?  A business?  An event? A product?)**

  Each row in the data provides relavant information about the patient.


**5. Is this a classification or regression problem?**

  This is a classification problem


**6. How many features does the data have?**

This data has 18 features.

**7. How many rows are in the dataset?**

There are 418 rows in the dataset.

**8. What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?** 

missing data could present a challenge.   The number of 'yes' 'no' features could present an issue if not handled correctly.

###**Dataset 2**

###Uploading Data:

In [10]:
filename = ("/content/drive/MyDrive/Colab Notebooks/healthcare-dataset-stroke-data.csv")
data2 = pd.read_csv(filename)
data2.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

In [11]:
data2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5110 entries, 0 to 5109
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5110 non-null   int64  
 1   gender             5110 non-null   object 
 2   age                5110 non-null   float64
 3   hypertension       5110 non-null   int64  
 4   heart_disease      5110 non-null   int64  
 5   ever_married       5110 non-null   object 
 6   work_type          5110 non-null   object 
 7   Residence_type     5110 non-null   object 
 8   avg_glucose_level  5110 non-null   float64
 9   bmi                4909 non-null   float64
 10  smoking_status     5110 non-null   object 
 11  stroke             5110 non-null   int64  
dtypes: float64(3), int64(4), object(5)
memory usage: 479.2+ KB


**1. Source of data:** https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset

**2. Brief description of data:**

  This dataset is used to predict whether a patient is likely to have a stroke
(target) 

  based on information about the patient (features) such as age, gender, bmi, etc.


**3. What is the target?** 

'stroke'

**4. What does one row represent? (A person?  A business?  An event? A product?)**

  Each row in the data provides relavant information about the patient.


**5. Is this a classification or regression problem?**

  This is a classification problem


**6. How many features does the data have?**

This data has 11 features.

**7. How many rows are in the dataset?**

There are 5,110 rows in the dataset.

**8. What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?** 

I do not foresee any major issues with this dataset, maybe the number of rows could present an issue.